### DISCUSSION TITLE
Use "Select" instead of "Textfield" in CRUD table

### DISCUSSION THREAD

**Discussion by Yitavg** [KevinVandy/material-react-table#304] [Q&A]
Hi,

I want to use CRUD in my table. But when adding new row, all the inputs are Textfields.
Using "muiTableBodyCellEditTextFieldProps" only creates a select component during cell edit, but I want it to be a select on creating a new row.

Also, when clicking on the "Discord" icon in the docs I get sent to "invalid invitation" page.

Thanks!

**Comment by KevinVandy** [comment]
I just had a couple people test the discord link and it is working perfectly fine?

Is this the editing example you are referring to? https://www.material-react-table.dev/?path=/story/features-editing-examples--editing-customize-input

After creating a new row, you can call the `table.setEditingRow(row)` somewhere to set that row as the row being edited. Though this might differ depending on what `editingMode` you are using.

If you want a select to always be present, consider just overriding the `Cell` render with the select component.

**Reply by Yitavg** [reply]
> I just had a couple people test the discord link and it is working perfectly fine?
> 
> Is this the editing example you are referring to? https://www.material-react-table.dev/?path=/story/features-editing-examples--editing-customize-input
> 
> After creating a new row, you can call the `table.setEditingRow(row)` somewhere to set that row as the row being edited. Though this might differ depending on what `editingMode` you are using.
> 
> If you want a select to always be present, consider just overriding the `Cell` render with the select component.

Thank you for the fast reply.
In the example that you show the "Select" component is only for the row edit, but I wish to use the Select when creating a new row.
I the example "select state" is only available on edit but when creating a new row "state" is a regular Textfield.

The discord link still says "invite invalid"  :(

**Reply by KevinVandy** [reply]
Take a screenshot of which discord link you're trying. Your country or VPN or company network might also just be blocking it
